### ISSUE TITLE
U value inverted?

### ISSUE THREAD

**Issue by idavydov** [Accio/BioQC#25]
Hi David,

I noticed that `U` value computed by `wmwTest` seems to be inverted. This also results in production of inverted `r` and `f` values.

Please find a reprex below.

What do you think is the right solution for this?
1. Update only behavior of `r` and `r` scores?
2. Also update the `U` score computation and potentially break compatibility? E.g., with some warning messages?
3. Add a flag `inverted_U=TRUE`, which is default?
4. Something else?

``` r
library(BioQC)
#> Loading required package: Biobase
#> Loading required package: BiocGenerics
#> Loading required package: parallel
#> 
#> Attaching package: 'BiocGenerics'
#> The following objects are masked from 'package:parallel':
#> 
#>     clusterApply, clusterApplyLB, clusterCall, clusterEvalQ,
#>     clusterExport, clusterMap, parApply, parCapply, parLapply,
#>     parLapplyLB, parRapply, parSapply, parSapplyLB
#> The following objects are masked from 'package:stats':
#> 
#>     IQR, mad, sd, var, xtabs
#> The following objects are masked from 'package:base':
#> 
#>     anyDuplicated, append, as.data.frame, basename, cbind, colnames,
#>     dirname, do.call, duplicated, eval, evalq, Filter, Find, get, grep,
#>     grepl, intersect, is.unsorted, lapply, Map, mapply, match, mget,
#>     order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,
#>     rbind, Reduce, rownames, sapply, setdiff, sort, table, tapply,
#>     union, unique, unsplit, which, which.max, which.min
#> Welcome to Bioconductor
#> 
#>     Vignettes contain introductory material; view with
#>     'browseVignettes()'. To cite Bioconductor, see
#>     'citation("Biobase")', and for packages 'citation("pkgname")'.
set.seed(10)
x <- rnorm(10, 1)
y <- rnorm(100, 0)

wilcox.test(x, y, alternative='greater')
#> 
#>  Wilcoxon rank sum test with continuity correction
#> 
#> data:  x and y
#> W = 697, p-value = 0.02052
#> alternative hypothesis: true location shift is greater than 0
# p.value is computed correctly
wmwTest(c(x, y), seq_along(x), valType='p.greater')
#> [1] 0.02052079
# U/W value seems to be inverted
wmwTest(c(x, y), seq_along(x), valType='U')
#> [1] 303
# let's invert and check
wmwTest(c(y, x), seq_along(y), valType='U')
#> [1] 697
```

<sup>Created on 2020-12-07 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

**Comment by idavydov** [comment]
#26 corrects inverted `u` and `r`, but not changing the `U` behavior.

**Comment by idavydov** [comment]
In fact, we just had a discussion with @JAttig, and he pointed out that there are U1 and U2.
Maybe the cleanest way is to:

- Raise a warning for `valType == "U"`
- Add `"U1"` and `"U2"` explicitly.

**Comment by Accio** [comment]
thanks, good points, fixed in 4c793cd.

No warning (U was not really used so far), but U used to point to U2, now it points to U1. U1 and U2 are options, too.

I hope that makes sense. Tests about U1 and U2 remain to be implemented.
